### PR TITLE
loosens the solidus core version to < 2.0

### PR DESCRIPTION
**Pull request by sherryson**
this is required for apps using this gem to upgrade to higher versions of Solidus (e.g. 1.4.2)
also removes Solidus from the Gemfile - seems odd to be based off the master branch and was preventing me from being able to bundle (lots of conflicts)

### CODE REVIEW

**Comment by sherryson** [comment]
@braidn not sure if you might have any insights here, but i'm really struggling to run the specs. specifically can't run `bundle exec rake test_app` successfully

**Comment by braidn** [comment]
@sherryson how are you installing the gems here? I recommend something like:

`bundle install --path=.bundle`

This will build a clean set of Gems locally for just this project. Then let's see what the bundle exec rake test_app fails with? What is the current failure output?

**Comment by sherryson** [comment]
@braidn oooh ok. i was just doing `bundle install`

currently `bundle exec rake test_app` gives me 
```
Generating dummy Rails application...
Setting up dummy database...
bundle exec rake db:migrate VERBOSE=false
rake aborted!
NameError: uninitialized constant Spree::UserRegistrationsController
/Users/sherryson/src/solidus_social/app/controllers/spree/user_registrations_controller_decorator.rb:1:in `<top (required)>'
/Users/sherryson/src/solidus_social/lib/solidus_social/engine.rb:21:in `block in activate'
/Users/sherryson/src/solidus_social/lib/solidus_social/engine.rb:20:in `glob'
/Users/sherryson/src/solidus_social/lib/solidus_social/engine.rb:20:in `activate'
/Users/sherryson/src/solidus_social/spec/dummy/config/environment.rb:5:in `<top (required)>'
/Users/sherryson/.rbenv/versions/2.3.4/bin/bundle:22:in `load'
/Users/sherryson/.rbenv/versions/2.3.4/bin/bundle:22:in `<main>'
Tasks: TOP => db:migrate => environment
(See full trace by running task with --trace)
rake aborted!
Command failed with status (1): [bundle exec rake db:migrate VERBOSE=false...]
/Users/sherryson/src/solidus_social/Rakefile:14:in `block in <top (required)>'
/Users/sherryson/.rbenv/versions/2.3.4/bin/bundle:22:in `load'
/Users/sherryson/.rbenv/versions/2.3.4/bin/bundle:22:in `<main>'
Tasks: TOP => common:test_app
(See full trace by running task with --trace)
```

so i tried creating a db/running migrations separately outside of this rake task but couldn't..

EDIT:
forgot to mention the first error with that rake task actually says we need deface in the Gemfile, so i'll add that to this PR too. looking at [solidus_auth_devise](https://github.com/solidusio/solidus_auth_devise), deface needs to be added to the Gemfile for versions of Solidus < 2.5

**Comment by sherryson** [comment]
got a little help from @braidn (thank you!)

following up on my last comment, we do need Solidus in the Gemfile, since `solidus-core` is only added as a runtime dependency, but instead of using the `master` branch, using the v1.4.2 is probably the right move. 

now the specs are failing (mainly because of older dependencies) and will fix and update here as well

File: spree_social.gemspec [hunk 1]
```diff
@@ -21,7 +21,7 @@ Gem::Specification.new do |s|
   s.require_path = 'lib'
   s.requirements << 'none'
 
-  s.add_runtime_dependency 'solidus_core', [">= 1.1.0.pre", "<= 1.4.0"]
+  s.add_runtime_dependency 'solidus_core', [">= 1.1.0.pre", "<= 2.0"]
```
Comment (ktheory) [inline spree_social.gemspec]:
```suggestion
  s.add_runtime_dependency 'solidus_core', [">= 1.1.0.pre", "< 2.0"]
```

Did you mean strictly `< 2.0` so we could go to v1.4.2 now, and revisit before 2.0?

If you want to allow v2.0, we should go up to the latest tiny version, `2.0.3`.

**Review by ktheory — COMMENTED** [review]
@braidn / @sherryson Is there background on why the constraint on `solids_core` is so pessimistic? Looking through the git history, it's been:

`<= 1.4.0`, `<= 1.3.0.alpha`, and `< 1.2.0`.

Was this out of an abundance of caution to force devs to bump the version before upgrading? Does solidus_social use private APIs we expect to break between point versions?

In short, is there a sufficient and more flexible version constraint than making it a bit wider each time we need to upgrade?

File: spree_social.gemspec (hunk 1 above)
Comment (braidn) [reply spree_social.gemspec]: I believe this work is to get this 💎 working with the 1.4 mainline ? Thereby doing incremental bumps to get us there. We also brought the gem version of Solidus in the Gemfile to 1.4.2 so we could be assured our test suite was passing with a 1.4 version of the app.

This causes micro updates which I agree are a little tiring but, due to how large our App is and how different it is from 1.4, being conservative here made sense. Maybe being hyperconservative wasn’t the best move?

**Comment by sherryson** [comment]
@ktheory unfortunately no... we couldn't figure out a way to fix the specs so instead i'll be QA-ing it through the main app. do you think it's worthwhile to add a feature spec for this in v2 while the specs in this gem aren't working?

**Comment by ktheory** [comment]
> do you think it's worthwhile to add a feature spec for this in v2 while the specs in this gem aren't working?

Let's move that discussion to a private v2 PR.

**Comment by sherryson** [comment]
thanks @DanielWright for fixing the specs!

@ktheory @braidn do you think i can merge this into master now?

File: spree_social.gemspec [hunk 2]
```diff
@@ -21,7 +21,7 @@ Gem::Specification.new do |s|
   s.require_path = 'lib'
   s.requirements << 'none'
 
-  s.add_runtime_dependency 'solidus_core', [">= 1.1.0.pre", "<= 1.4.0"]
+  s.add_runtime_dependency 'solidus_core', [">= 1.1.0.pre", "< 2.0"]
```
Comment (ktheory) [inline spree_social.gemspec:24]: Aces ♠️

**Review by ktheory — COMMENTED** [review]
LGTM. 👍

**Pull request merged**